bot/exts/utils/eval.py

```diff
@@ -0,0 +1,252 @@
+
+            stats option displays more information on execution consumption
+            wrapped allows you to not put main function in some languages
+
+            <code> may be normal code, but also an attached file, or a link from  \
+            [hastebin](https://hastebin.com) or [Github gist](https://gist.github.com)
+            If you use a link, your command must end with this syntax: \
+            `link=<link>` (no space around `=`)
+            for instance : `!eval python link=https://hastebin.com/gurkbot.py`
+
+            If the output exceeds 40 lines or Discord max message length, it will be put
+            in a new hastebin and the link will be returned.
+            ''',
+        brief='Execute code in a given programming language',
+        name="eval",
+        aliases=("e",)
+    )
+    async def eval_command(self, ctx: Context, language: str, *, code: str = '') -> Optional[Message]:
+        """
+        Evaluate code, format it, and send the output to the corresponding channel.
+
+        Return the bot response.
+        """
+        logger.info(f"Received code from {ctx.author} for evaluation:\n{code}")
+
+        options = {
+            '--stats': False,
+            '--wrapped': False
+        }
+        lang = language.strip('`').lower()
+        options_amount = len(options)
+
+        # Setting options and removing them from the beginning of the command
+        # options may be separated by any single whitespace, which we keep in the list
+        code = re.split(r'(\s)', code, maxsplit=options_amount)
+
+        for option in options:
+            if option in code[:options_amount * 2]:
+                options[option] = True
+                i = code.index(option)
+                code.pop(i)
+                code.pop(i)  # remove following whitespace character
+
+        code = ''.join(code)
+
+        compiler_flags = []
+        command_line_options = []
+        args = []
+        inputs = []
+
+        lines = code.split('\n')
+        code = []
+        for line in lines:
+            if line.startswith('input '):
+                inputs.append(' '.join(line.split(' ')[1:]).strip('`'))
+            elif line.startswith('compiler-flags '):
+                compiler_flags.extend(line[15:].strip('`').split(' '))
+            elif line.startswith('command-line-options '):
+                command_line_options.extend(line[21:].strip('`').split(' '))
+            elif line.startswith('arguments '):
+                args.extend(line[10:].strip('`').split(' '))
+            else:
+                code.append(line)
+
+        inputs = '\n'.join(inputs)
+        code = '\n'.join(code)
+        text = None
+
+        async with ctx.typing():
+            if ctx.message.attachments:
+                # Code in file
+                file = ctx.message.attachments[0]
+                if file.size > 20000:
+                    return await ctx.send("File must be smaller than 20 kio.")
```

Comment:
```suggestion
                    await ctx.send("File must be smaller than 20 kio.")
                    return
```
I generally like doing it like this, but both ways are probably fine.